### PR TITLE
NR46306: Lighttpd-new integration

### PR DESCRIPTION
**Pull request by dikshab0202**
Ticket: [NR-46306](https://issues.newrelic.com/browse/NR-46306)
Description: Lighttpd - New integration

Preview image:
image

# Summary

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

## Pre merge checklist

- [ ] Did you check you NRQL syntax? - Does it work?
- [ ] Did you include an InstallPlan and Documentation reference?
- [ ] Are all documentation links publically accessible?
- [ ] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)? 
- [ ] Did you check your descriptive content for spelling and grammar errors?

### Dashboards 

- [ ] Does the PR contain a screenshot for each of your dashboards?
- [ ] Do your screenshots show data?
- [ ] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [ ] Did you check that your alerts actually work?

### CODE REVIEW

**Comment by dikshab0202** [comment]
Waiting for Lighttpd doc url.

File: alert-policies/lighttpd/AverageNumberOfConnections.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Average Number of Connections
+
+description: |+
+  This alert is triggered when the average number of connections are above 6000 for atleast 2 minutes.
```
Comment (paperclypse) [inline alert-policies/lighttpd/AverageNumberOfConnections.yml:4]:
```suggestion
  This alert is triggered when the average number of connections are above 6000 for at least 2 minutes.
```

File: alert-policies/lighttpd/AverageNumberOfConnections.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    operator: ABOVE
+    # Value that triggers a violation
+    threshold: 6000
+    # Time in seconds; 120 - 3600
+    thresholdDuration: 120
+    # How many data points must be in violation for the duration
+    thresholdOccurrences: ALL
+
+  - priority: WARNING
+    # Operator used to compare against the threshold.
```
Comment (paperclypse) [inline alert-policies/lighttpd/AverageNumberOfConnections.yml:25]: Why do some of these comments end in periods and not others. I prefer consistency :)

File: alert-policies/lighttpd/AvgNumberOfRequestsFor5sec.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: Average Number of Requests for 5 Seconds
+
+description: |+
+  This alert is triggered when the average number of requests for 5 seconds are above 6000 for atleast 5 minutes.
```
Comment (paperclypse) [inline alert-policies/lighttpd/AvgNumberOfRequestsFor5sec.yml:4]:
```suggestion
  This alert is triggered when the average number of requests for 5 seconds are above 6000 for at least 5 minutes.
```

File: alert-policies/lighttpd/HighMemoryUsage.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+name: High Memory Usage
+
+description: |+
+  This alert is triggered when the memory usage is above 90% for atleast 5 minutes.
```
Comment (paperclypse) [inline alert-policies/lighttpd/HighMemoryUsage.yml:4]:
```suggestion
  This alert is triggered when the memory usage is above 90% for at least 5 minutes.
```

File: install/lighttpd/install.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+id: lighttpd
+name: lighttpd
+title: Lighttpd
+description: |
+  To collect data and deliver it to New Relic, integrate your Lighttpd processes with our Infrastructure agent and Flex configuration.
```
Comment (paperclypse) [inline install/lighttpd/install.yml:5]:
```suggestion
  To collect data and deliver it to New Relic, integrate your lighttpd processes with our Infrastructure agent and Flex configuration.
```

Based on its website, lighttpd is always lowecase

File: quickstarts/lighttpd/config.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+
+slug: lighttpd
+description: |
+  ## Comprehensive monitoring quickstart for Lighttpd
+
+  Lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure and adaptable.
```
Comment (paperclypse) [inline quickstarts/lighttpd/config.yml:6]:
```suggestion
  Lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure, and adaptable.
```

This is a great intro sentence! We prefer the Oxford (or serial) comma though :)

File: quickstarts/lighttpd/config.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  ### Why monitor Lighttpd?
```
Comment (paperclypse) [inline quickstarts/lighttpd/config.yml:8]:
```suggestion
  ### Why monitor your lighttpd server?
```

File: install/lighttpd/install.yml (hunk 5 above)
Comment (paperclypse) [reply install/lighttpd/install.yml:5]: Also, did you know lighttpd is pronounced _lighty_? :)

File: quickstarts/lighttpd/config.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  Lighttpd monitoring provides important information to help the developers construct a complete picture of a web server's performance. This data comprises uptime, network in bytes and packets, number of connections and more.
```
Comment (paperclypse) [inline quickstarts/lighttpd/config.yml:10]:
```suggestion
  Lighttpd monitoring provides important information to help you construct a complete picture of your web server's performance. The performance data includes uptime, network in bytes and packets, number of connections, and more.
```

File: quickstarts/lighttpd/config.yml [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+  ### What’s included?
+
+  The Lighttpd monitoring quickstart from New Relic provides immediate full-stack observability:
```
Comment (paperclypse) [inline quickstarts/lighttpd/config.yml:14]:
```suggestion
  When you monitor your lighttpd server withNew Relic, you get immediate full-stack observability, including:
```

File: quickstarts/lighttpd/config.yml [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  - Alerts (average number of connections, average number of requests, high memory usage).
+  - Dashboards (uptime, connection state handle request, connection state still alive, networks, requests per second and more).
+  - Be proactive in monitoring potential issues and resolving them as soon as possible.
```
Comment (paperclypse) [inline quickstarts/lighttpd/config.yml:17]: I'd delete this last bullet. It doesn't seem very helpful or informative.

If we must keep it, I'd rewrite it to:

"Deeper insight into your server's performance issues."

File: quickstarts/lighttpd/config.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+summary: |
+   With our Infrastructure agent and Flex configuration setup, you can monitor and validate your Lighttpd processes. Right out of the box, analyze and capture performance data.
```
Comment (paperclypse) [inline quickstarts/lighttpd/config.yml:20]:
```suggestion
   With our infrastructure agent and Flex configuration setup, you can monitor and validate your lighttpd processes. Right out of the box, analyze and capture your server's performance data.
```

**Review by paperclypse — DISMISSED** [review]
I left some comments and suggested edits.

File: quickstarts/lighttpd/config.yml (hunk 11 above)
Comment (dikshab0202) [reply quickstarts/lighttpd/config.yml:20]: Flex is an application-agnostic, all-in-one [New Relic integration](https://docs.newrelic.com/docs/integrations) with which you can instrument any app that exposes metrics over a standard protocol (HTTP, file, shell) in a standard format.
 Sharing [github url](https://github.com/newrelic/nri-flex) for the better understanding.

File: install/lighttpd/install.yml (hunk 5 above)
Comment (dikshab0202) [reply install/lighttpd/install.yml:5]: Yes, Shawn.

Comment (paperclypse) [reply install/lighttpd/install.yml]: I didn't :) I was very surprised!

File: quickstarts/lighttpd/config.yml [hunk 12]
```diff
@@ -0,0 +1,39 @@
+
+slug: lighttpd
+description: |
+  ## Comprehensive monitoring for your lighttpd server
+
+  lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure, and adaptable.
```
Comment (paperclypse) [inline quickstarts/lighttpd/config.yml:6]: lighttpd should still be capitalized when it's the start of a sentence.

File: quickstarts/lighttpd/config.yml (hunk 11 above)
Comment (paperclypse) [reply quickstarts/lighttpd/config.yml]: Thank you!

File: quickstarts/lighttpd/config.yml (hunk 12 above)
Comment (dikshab0202) [reply quickstarts/lighttpd/config.yml]: okay @paperclypse . I applied the above changes. 
If possible can you please once check [lighttpd page](https://www.lighttpd.net/), I found there it started with small letter in the starting of a sentence as well.

Comment (paperclypse) [reply quickstarts/lighttpd/config.yml]: For the start of a sentence, I'd prefer to override their style choice (lowercase at start of sentence) with ours (always capitalizing the start of a sentence). I looked through their docs and they're not consistent on this point, so I feel pretty okay with capitalizing it at the start of a sentence.

Comment (dikshab0202) [reply quickstarts/lighttpd/config.yml]: Okay @paperclypse, no problem. I have applied all the changes and pushed. Kindly review and let me know if any more suggestions or changes required.
Thank you.

Comment (paperclypse) [reply quickstarts/lighttpd/config.yml]: Looks good to me, thanks!

**Review by paperclypse — APPROVED** [review]
Looks good!

**Comment by brammerl** [comment]
Hello, We are no longer accepting install plans as a part of the quickstart schema. We are instead moving to use data sources. You can either refer to an [already existing data source](https://github.com/newrelic/newrelic-quickstarts/tree/main/data-sources) in this repository or[ make your own](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1) community data source.

If this applies to your PR, please make the proper update to your quickstart's config to reference only data sources.

**Comment by dikshab0202** [comment]
> Hello, We are no longer accepting install plans as a part of the quickstart schema. We are instead moving to use data sources. You can either refer to an [already existing data source](https://github.com/newrelic/newrelic-quickstarts/tree/main/data-sources) in this repository or[ make your own](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1) community data source.
> 
> If this applies to your PR, please make the proper update to your quickstart's config to reference only data sources.

Thank you @brammerl . I have updated this PR by adding datasources files and removed install plan. Could you please review and let me know if any changes are required.

**Review by pkosanam — APPROVED** [review]
looks good

File: data-sources/lighttpd/config.yml [hunk 13]
```diff
@@ -0,0 +1,22 @@
+id: lighttpd
```
Comment (pkosanam) [inline data-sources/lighttpd/config.yml:1]: Looks good..
There are few more fields which are to be added.

Request you to kindly follow this contribution guide
guide: https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#data-sources-1
Thank you 😃

**Review by pkosanam — DISMISSED** [review]
looks good

File: quickstarts/lighttpd/config.yml [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+  Lighttpd is an open-source web server designed for high-performance situations while staying standards-compliant, secure, and adaptable.
+
+  ### Why monitor your lighttpd server?
+
+  Lighttpd monitoring provides important information to help you construct a complete picture of your web server's performance. This performance data includes uptime, network in bytes and packets, number of connections, and more.
+
+  ### What’s included?
+
+  When you monitor your lighttpd server with New Relic, you get immediate full-stack observability, including:
+  - Be alerted on things like average number of connections, average number of requests and high memory usage.
```
Comment (seemantk) [inline quickstarts/lighttpd/config.yml:15]: Be alerted on things like -> Alerts on

File: data-sources/lighttpd/config.yml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+displayName: Lighttpd
+description: |
+  ## Comprehensive monitoring for your lighttpd server
```
Comment (brammerl) [inline data-sources/lighttpd/config.yml:4]: Data source descriptions don't support markdown and are expected to be a short description. Please come up with a short/concise description for this data source

File: quickstarts/lighttpd/config.yml [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+  - Use dashboards to monitor crucial performance metrics, such as uptime, connection state handle request, connection state still alive, networks, requests per second, and more.
```
Comment (seemantk) [inline quickstarts/lighttpd/config.yml:16]: Use dashboards -> Dashboards
performance metrics, -> performance metrics

File: quickstarts/lighttpd/config.yml [hunk 17]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (brammerl) [inline quickstarts/lighttpd/config.yml:1]: Let's remove this space

File: quickstarts/lighttpd/config.yml [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+  Lighttpd monitoring provides important information to help you construct a complete picture of your web server's performance. This performance data includes uptime, network in bytes and packets, number of connections, and more.
+
+  ### What’s included?
+
+  When you monitor your lighttpd server with New Relic, you get immediate full-stack observability, including:
+  - Be alerted on things like average number of connections, average number of requests and high memory usage.
+  - Use dashboards to monitor crucial performance metrics, such as uptime, connection state handle request, connection state still alive, networks, requests per second, and more.
+
+summary: |
+   With our infrastructure agent and Flex configuration setup, you can monitor and validate your lighttpd processes. Right out of the box, analyze and capture your server's performance data.
```
Comment (seemantk) [inline quickstarts/lighttpd/config.yml:19]: "lighttpd processes. Right out of the box," -> "lighttpd processes right out of the box:"

File: quickstarts/lighttpd/config.yml [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+  - name: Lighttpd
+    description: |
+      To collect data and deliver it to New Relic, integrate your lighttpd processes with our infrastructure agent and Flex configuration.
+    url: >-
+      https://docs.newrelic.com/docs/infrastructure/host-integrations/host-integrations-list/lighttpd-integration/
+installPlans:
+  - guided-install
+dataSourceIds:
+  - lighttpd
+keywords:
```
[the remark this answers is not in the document]
Comment (dikshab0202) [reply quickstarts/lighttpd/config.yml:38]: Hi @brammerl , Thank you for the information. I did these changes. Kindly review and let me know if something incorrect.

File: quickstarts/lighttpd/config.yml (hunk 14 above)
Comment (dikshab0202) [reply quickstarts/lighttpd/config.yml]: @seemantk , Changes have been completed. Kindly review and let me know if something incorrect.
Thank you..

**Review by pkosanam — APPROVED** [review]
looks good

**Review by brammerl — APPROVED** [review]

**Pull request merged**